iothub-manager/app/com/microsoft/azure/iotsolutions/iothubmanager/services/Deployments.java

```diff
@@ -0,0 +1,310 @@
+                                    return new DeploymentServiceModel(config);
+                                } catch (InvalidInputException e) {
+                                    throw new CompletionException(e);
+                                }
+                            })
+                            .sorted(Comparator.comparing(DeploymentServiceModel::getName))
+                            .collect(Collectors.toList());
+            return CompletableFuture.supplyAsync(() -> new DeploymentServiceListModel(serviceModelDeployments));
+        } catch (IOException | IotHubException e) {
+            final String message = "Unable to list deployments";
+            log.error(message, e);
+            throw new ExternalDependencyException(message, e);
+        }
+
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public CompletionStage<DeploymentServiceModel> getAsync(String id, boolean includeDeviceStatus) throws
+            ExternalDependencyException {
+        try {
+            final Configuration deployment = this.registry.getConfiguration(id);
+
+            if (deployment == null) {
+                throw new CompletionException(
+                    new ResourceNotFoundException(String.format("No deployment with id %s found for %s hub.",
+                                                                id, this.ioTHubHostName)));
+            }
+
+            final DeploymentServiceModel result;
+            try {
+                result = new DeploymentServiceModel(deployment);
+            } catch (InvalidInputException e) {
+                throw new CompletionException(e);
+            }
+
+            if (includeDeviceStatus) {
+                Map<String, DeploymentStatus> deviceStatuses = null;
+
+                try {
+                    deviceStatuses = this.getDeviceStatuses(id);
+                } catch(Exception ex) {
+                    log.error("Unable to retrieve device statuses for deployment " + id, ex);
+                }
+
+                result.getDeploymentMetrics().setDeviceStatuses(deviceStatuses);
+            }
+            return CompletableFuture.supplyAsync(() -> result);
+        } catch (IotHubNotFoundException e) {
+            throw new CompletionException(
+                    new ResourceNotFoundException(String.format("No deployment with id %s found for %s hub.",
+                            id, this.ioTHubHostName)));
+        } catch (IOException | IotHubException e) {
+            final String message = "Unable to list deployments";
+            log.error(message, e);
+            throw new ExternalDependencyException(message, e);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public CompletionStage<DeploymentServiceModel> createAsync(DeploymentServiceModel deployment) throws
+            InvalidInputException {
+        this.verifyDeploymentArgument(DEVICE_GROUP_ID_PARAM, deployment.getDeviceGroupId());
```

Comment:
nit: may need more consistence to use Argument/Parameter for method name and parameter name.